libafl/src/stages/tuneable.rs

```diff
@@ -227,43 +239,50 @@ where
     }
 }
 
-impl<E, EM, M, Z> TuneableMutationalStage<E, EM, Z::Input, M, Z>
+impl<E, EM, I, M, Z> TuneableMutationalStage<E, EM, I, M, Z>
 where
     E: UsesState<State = Z::State>,
     EM: UsesState<State = Z::State>,
-    M: Mutator<Z::Input, Z::State>,
+    M: Mutator<I, Z::State>,
     Z: Evaluator<E, EM>,
-    Z::State: HasClientPerfMonitor + HasCorpus + HasRand + HasMetadata,
+    Z::State: HasClientPerfMonitor + HasCorpus + HasRand + HasNamedMetadata + HasMetadata,
 {
-    /// Creates a new default mutational stage
+    /// Creates a new default tuneable mutational stage
     #[must_use]
     pub fn new(state: &mut Z::State, mutator: M) -> Self {
-        Self::transforming(state, mutator)
+        Self::transforming(state, mutator, DEFAULT_TUNEABLE_MUTATIONAL_STAGE_NAME)
+    }
+
+    /// Crates a new tuneable mutational stage with the given name
+    pub fn with_name(state: &mut Z::State, mutator: M, name: &str) -> Self {
+        Self::transforming(state, mutator, name)
     }
-}
 
-impl TuneableMutationalStage<(), (), (), (), ()> {
     /// Set the number of iterations to be used by this mutational stage
-    pub fn set_iters<S: HasMetadata>(state: &mut S, iters: u64) -> Result<(), Error> {
-        set_iters(state, iters)
+    pub fn set_iters<S: HasNamedMetadata>(&self, state: &mut S, iters: u64) -> Result<(), Error> {
+        set_iters(state, iters, &self.name)
     }
 
     /// Get the set iterations
-    pub fn iters<S: HasMetadata>(state: &S) -> Result<Option<u64>, Error> {
-        get_iters(state)
+    pub fn iters<S: HasNamedMetadata>(&self, state: &S) -> Result<Option<u64>, Error> {
```

Comment:
Maybe have a `iters_named` fn then?

Reply:
IMO using named is just fine, this is called once per scheduled input IIRC, so not really an hot path.